internal/obsreportconfig/obsreportconfig.go

```diff
@@ -20,9 +20,21 @@ import (
 	"go.opencensus.io/tag"
 
 	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/internal/obsreportconfig/obsmetrics"
 )
 
+const (
+	// UseOtelForInternalMetricsfeatureGateID is the feature gate ID that controls whether the collector uses open
+	// telemetrySettings for internal metrics.
+	UseOtelForInternalMetricsfeatureGateID = "telemetry.useOtelForInternalMetrics"
+)
+
+// UseOtelForMetrics controls whether the Collector should use Otel for Collector's internal instrumentation.
```

Comment:
nit: this method doesn't control anything but reports it 

```suggestion
// UseOtelForMetrics returns whether the Collector should use Otel for Collector's internal instrumentation.
```

Reply:
I don't think this function is even needed since the feature gate ID constant is also exported.  It's easy enough for consumers to check the gate themselves and also hopefully avoids hiding the fact that a gate is being checked so it isn't called in a tight loop accidentally.

Reply:
Just removed it.